cmake/IBAMRTesting.cmake

```diff
@@ -0,0 +1,285 @@
+
+# Find numdiff
+find_program(NUMDIFF_EXECUTABLE NAMES numdiff HINTS ${NUMDIFF_ROOT} PATH_SUFFIXES bin)
+if ("${NUMDIFF_EXECUTABLE}" STREQUAL "NUMDIFF_EXECUTABLE-NOTFOUND")
+  message(WARNING "\
+The configuration script was not able to locate numdiff. If you want to run \
+the test suite you will need to either edit attest.conf, specify the path to \
+numdiff to attest, or rerun CMake with the argument NUMDIFF_ROOT specifying \
+numdiff's root installation directory.")
+  # clear the value so that attest.conf doesn't contain an invalid path
+  set(NUMDIFF_EXECUTABLE "")
+  set(IBAMR_ENABLE_TESTING FALSE)
+endif ()
+
+set(IBAMR_ALLOW_OVERSUBSCRIBED OFF "Allow tests to be added if they are oversubscribed")
+mark_as_advanced(
+  NUMDIFF_EXECUTABLE
+  IBAMR_TEST_DIR
+  IBAMR_TEST_TIMEOUT
+  IBAMR_ALLOW_OVERSUBSCRIBED)
+
+function (_ibamr_parse_test name)
+  set(n_mpirun 1)
+  set(n_restart 0)
+  set(expect_error FALSE)
+
+  string(REPLACE "." ";" name_list "${name}")
+  foreach (override ${name_list})
+    string(REPLACE "=" ";" _x ${override})
+    list(LENGTH _x _x_len)
+    if (${_x_len} EQUAL 2)
+      list(GET _x 0 _x_key)
+      list(GET _x 1 _x_value)
+      if (_x_key STREQUAL "mpirun")
+        set(n_mpirun ${_x_value})
+      elseif (_x_key STREQUAL "expect_error")
+        string(TOUPPER "${_x_value}" value)
+        set("expect_error" ${value})
+      elseif (_x_key STREQUAL "restart")
+        set("restart" ${_x_value})
+      else (_x_key STREQUAL "restart")
+        message(WARNING "Encountered unrecognized override ${_x_key}")
+      endif ()
+      message(STATUS "${_x_key} ${_x_value}")
+    endif ()
+  endforeach ()
+
+  set(n_mpirun ${n_mpirun} PARENT_SCOPE)
+  set(n_restart ${n_restart} PARENT_SCOPE)
+  set(expect_error ${expect_error} PARENT_SCOPE)
+endfunction()
+
+#[[!md
+
+ibamr_add_testss(
+  base_name
+  DIRECTORY
+    TestDirectory
+)
+
+]]#
+function (ibamr_add_tests name)
+  cmake_parse_arguments(test
+    ""
+    "DIRECTORY"
+    "LIBRARIES;SOURCES" ${ARGN})
+
+  if (NOT test_DIRECTORY)
+    message(FATAL_ERROR "Missing DIRECTORY for test")
+  endif ()
+
+  if (NOT IBAMR_ENABLE_TESTING)
+    return ()
+  endif ()
+
+  #message(CHECK_START "Adding tests: ${test_DIRECTORY}/${name}")
+  #list(APPEND CMAKE_MESSAGE_INDENT "  ")
+
+  # Add test source if it exists (may not because of _2d/_3d suffix)
+  if (EXISTS "${IBAMR_TEST_DIR}/${test_DIRECTORY}/${name}.cpp")
+    list(APPEND test_SOURCES "${IBAMR_TEST_DIR}/${test_DIRECTORY}/${name}.cpp")
+  endif ()
+
+  # If there aren't any sources, the this is not going to work...
+  if (NOT test_SOURCES)
+    message(FATAL_ERROR "No sources detected for test: ${test_DIRECTORY}/${name}")
+  endif ()
+
+  # Find all of the test inputs
+  file(GLOB input_files "${IBAMR_TEST_DIR}/${test_DIRECTORY}/${name}.*.input")
+  if (NOT input_files)
+    #list(POP_BACK CMAKE_MESSAGE_INDENT)
+    #message(CHECK_FAIL "no test inputs detected, skipping...")
```

Comment:
This was part of my question above related to writing out which tests are being added during configuration. I commented them out because they produced a lot of output but I wanted to get more guidance from you on what kind of output would be useful here, if any.

Reply:
I'd prefer to just not output anything.